features/Blocklist.php

```diff
@@ -142,7 +142,7 @@ private function _regenerateCache(): bool
      *
      * @return int|Exception 1 if the hash exists or an Exception if something failed
      */
-    private function _passwordCacheMiss(GenericCache $cache, string $passwordHash): bool
+    private function _passwordCacheMiss(GenericCache $cache, string $passwordHash): bool|Exception
```

Comment:
I got a fatal error while testing a password change from the user profile with all password complexity options checked On, but then couldn't reproduce it. `PHP Fatal error:  Uncaught TypeError: APP\plugins\generic\betterPassword\features\Blocklist::_passwordCacheMiss(): Return value must be of type Exception|bool, null returned in [...]/plugins/generic/betterPassword/features/Blocklist.php:157` Not that the original code would have fared any better...

Reply:
I've inspected a bit and the `fallback` function isn't called when the cache is generated.

Another problem, that I flagged is that the `_regenerateCache()` will be called whenever a "hash group" isn't found in the cache, this shouldn't happen, I'll create another issue for it.

Reply:
I've included `null` in the return.
p.s.: It's not worth to fight too much with these "cache" classes, they were removed in OJS 3.5.